src/dms/core/EdFi.DataManagementService.Core/OpenApi/OpenApiDocument.cs

```diff
@@ -0,0 +1,154 @@
+// SPDX-License-Identifier: Apache-2.0
+// Licensed to the Ed-Fi Alliance under one or more agreements.
+// The Ed-Fi Alliance licenses this file to you under the Apache License, Version 2.0.
+// See the LICENSE and NOTICES files in the project root for more information.
+
+using System.Text.Json.Nodes;
+using EdFi.DataManagementService.Core.ApiSchema;
+using EdFi.DataManagementService.Core.ApiSchema.Extensions;
+using Microsoft.Extensions.Logging;
+
+namespace EdFi.DataManagementService.Core.OpenApi;
+
+/// <summary>
+/// Provides information from a loaded ApiSchema.json document
+/// </summary>
+internal class OpenApiDocument(ILogger _logger)
+{
+    /// <summary>
+    /// Inserts exts from extension OpenAPI fragments into the _ext section of the corresponding
+    /// core OpenAPI endpoint.
+    /// </summary>
+    private void InsertExts(JsonObject extList, JsonNode openApiSpecification)
+    {
+        foreach ((string componentSchemaName, JsonNode? extObject) in extList)
+        {
+            if (extObject == null)
+            {
+                throw new InvalidOperationException(
+                    $"OpenAPI extension fragment has empty exts schema name '{componentSchemaName}'. Extension fragment validation failed?"
```

Comment:
Everywhere with "Extension fragment validation failed?" is an opportunity for validation with DMS-496. Noted that in ticket